src/components/Icon/Icon.jsx

```diff
@@ -72,9 +78,20 @@ const Icon = createClass({
 			viewBox: '0 0 16 16',
 			isBadge: false,
 			isDisabled: false,
+			isClickable: false,
 		};
 	},
 
+	onSelect(event) {
+		const { isClickable, isDisabled, onClick } = this.props;
```

Comment:
You're right to pluck out `onClick` but we should have some code like this:

```javascript
if (onClick) {
  onClick(event);
}
```

That way existing users who are currently relying on `onClick` to have specific arguments won't be broken. Instead we'll be introducing a new `onSelect` that has the `{ event, props }` object.